src/tiramisu_core.cpp

```diff
@@ -97,6 +97,26 @@ void codegen(const std::vector<tiramisu::buffer *> &arguments, const std::string
     fct->codegen(arguments, obj_filename, gen_architecture_flag);
 }
 
+bool check_legality_of_function()
+{
+    function *fct = global::get_implicit_function();
+    return fct->check_legality_for_function() ;
+}
+
+
+void performe_full_dependecy_analysis()
```

Comment:
Ok I see now the difference.

Reply:
a lot of these methods uses the function object's and it's attributes. But since the function should not be exposed to regular users, I duplicated these. so you will see the function methods where the real execution is done, and a code exposed in tiramisu:: which only gets the implicit function a call the needed method.

Reply:
Yes indeed I saw that wile reading the code.